src/general/PlayerInspectInfo.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Godot;
+
+/// <summary>
+///   A system that manages detecting what the player is pointing with the cursor.
+/// </summary>
+public class PlayerInspectInfo : Node
+{
+    /// <summary>
+    ///   The distance for detection.
+    /// </summary>
+    [Export]
+    public float RayastDistance = 1000;
+
+    private readonly List<RaycastResult> hits = new();
+    private readonly HashSet<RaycastResult> previousHits = new();
+
+    /// <summary>
+    ///   All inspectable entities the player is pointing.
+    /// </summary>
+    public IReadOnlyList<IInspectableEntity> InspectableEntities =>
```

Comment:
I think this should return an enumerable and leave it up to the caller to decide when they have to make a concrete list out of it.